tests/conftest.py

```diff
@@ -1649,25 +1653,68 @@ def collect_db_dump(request, duthosts):
 
-    yield
-    if "20191130" in duthost.os_version:
-        cur_cores = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
-    else:
-        cur_cores = duthost.shell('ls /var/core/')['stdout'].split()
+            duthost_dict[duthost]["pre_running_config"] = duthost.get_running_config()["ansible_facts"]["running_config"]
 
-    new_core_dumps = set(cur_cores) - set(pre_existing_cores)
-    # convert to list so print msg will not contain "set()"
-    new_core_dumps = list(new_core_dumps)
+    yield
 
-    if new_core_dumps:
-        pytest.fail("Core dumps found. Expected: %s Found: %s. Test failed. New core dumps: %s" % (len(pre_existing_cores),\
-            len(cur_cores), new_core_dumps))
+    if running_flag:
+        for duthost in duthost_dict:
+            if "20191130" in duthost.os_version:
+                cur_cores = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
+            else:
+                cur_cores = duthost.shell('ls /var/core/')['stdout'].split()
+            duthost_dict[duthost]["cur_cores"] = cur_cores
+
+            new_core_dumps = set(duthost_dict[duthost]["cur_cores"]) - set(duthost_dict[duthost]["pre_cores"])
+            # convert to list so print msg will not contain "set()"
+            new_core_dumps = list(new_core_dumps)
+
+            if new_core_dumps:
+                check_result["core_dump"]["failed"] = True
+                check_result["core_dump"]["corefile"] = new_core_dumps
+                logger.info("Core dumps found. Expected: %s Found: %s. New core dumps: %s" % (len(pre_existing_cores), len(cur_cores), new_core_dumps))
+
+            duthost_dict[duthost]["cur_running_config"] = duthost.get_running_config()["ansible_facts"]["running_config"]
+            # Check if the running config is modified after the test case running
+            different_config = duthost.compare_config(pre_running_config=duthost_dict[duthost]["pre_running_config"], cur_running_config=duthost_dict[duthost]["cur_running_config"])["ansible_facts"]["different_config"]
+            logger.info('Deferences between running config and bak golden config:\n{}'.format(different_config))
+            if different_config["different_config_cur"]:
+                check_result["config"]["failed"] = True
+                check_result["config"]["different_config"] = different_config
+
+            new_failed_results = [{key: value} for key, value in check_result.items() if value['failed']]
+            if new_failed_results:
+                # Reload with config before test running
+                logger.info("Dut health check results {}".format(new_failed_results))
+
+                logger.info("Reload with config before test running")
+                duthost.command("echo {} > {}".format(duthost_dict[duthost]["pre_running_config"], CONFIG_DB))
+                config_reload(duthost)
+                pytest.fail("Dut is not health after running the test case.")
```

Comment:
In case something went wrong, it will fail on the first duthost. Rest of the duthosts won't get a chance to run such checks.